flexmeasures/api/v1_3/implementations.py

```diff
@@ -144,19 +145,25 @@ def get_device_message_response(generic_asset_name_groups, duration):
                     return unknown_schedule("Scheduling job has an unknown status.")
                 schedule_start = job.kwargs["start"]
 
-            schedule_data_source_name = "Seita"
-            scheduler_source = DataSource.query.filter_by(
-                name=schedule_data_source_name, type="scheduling script"
-            ).one_or_none()
-            if scheduler_source is None:
-                return unknown_schedule(
-                    message + f'no data is known from "{schedule_data_source_name}".'
-                )
+            data_source_info = None
+            if job:
+                data_source_info = job.meta.get("data_source_info")
+            if data_source_info is None:
+                data_source_info = dict(
+                    name="Seita"
+                )  # TODO: change to raise later - all scheduling jobs now get full info
+            scheduler_sources = DataSource.query.filter_by(
+                type="scheduling script",
+                **data_source_info,
+            ).all()  # Might be more than one, e.g. per user
+            if len(scheduler_sources) == 0:
+                s_info = ",".join([f"{k}={v}" for k, v in data_source_info.items()])
+                return unknown_schedule(message + f"no data is known from [{s_info}].")
 
             power_values = sensor.search_beliefs(
                 event_starts_after=schedule_start,
                 event_ends_before=schedule_start + planning_horizon,
-                source=scheduler_source,
+                source=scheduler_sources[-1],
```

Comment:
Why only the last one? Also, without an `order_by` statement in the `DataSource` query, this may result in unexpected behaviour.

Reply:
I added ordering by version, to catch the source with the highest. Still just a heuristic, but the right one until we can expect data source information to be complete.
What could be surprising would be if two sources exist, with different users.

Reply:
I see. Perhaps this could be resolved by storing the DataSource id as job meta (possibly instead of the data source info dict)?

Reply:
I also added now that we save the actual data source ID. So in most cases, we know we get one exactly, and that other code can later be cleaned.

Reply:
Ah, so we both had the same idea 😃